packages/flutter_test/lib/src/binding.dart

```diff
@@ -1523,8 +1550,9 @@ class LiveTestWidgetsFlutterBinding extends TestWidgetsFlutterBinding {
         break;
       case TestBindingEventSource.device:
         if (deviceEventDispatcher != null) {
+          final PointerEvent localEvent = event.copyWith(position: globalToLocal(event.position));
```

Comment:
This line probably deserves a comment explaining why the position needs to be transformed.

Reply:
I added 
```dart
          // The pointer events received with this source has a global position
          // (see [handlePointerEventForSource]). Transform it to the local
          // coordinate space used by the testing widgets.
```